### PR TITLE
Fix for the Carbon isotope land-use change bug

### LINKED ISSUE #675: Enabling carbon isotopes changes answers for transient cases
### Brief summary of bug

Turning on C13 or C14 shouldn't change answers, and yet it does in certain cases.

### General bug information

**CTSM version you are using:** release-clm5.0.20

**Does this bug cause significantly incorrect results in the model's science?** Maybe?

**Configurations affected:** C13 or C14 turned on

### Details of bug

Adding use_c13=T or use_c14=T, to the SSPRCP126 compset 

### Important details of your setup / configuration so we can reproduce the bug

Saw this at first in the BSSP126 compset versus BSSP126cmip6. Can also see it in I compset cases.

My test case is: SMS_Ln20.f09_g17_gl4.ISSP126Clm50BgcCrop.cheyenne_intel.clm-crop

Where I've added the following:

``` csh
./xmlchange GET_REFCASE=TRUE,RUN_REFCASE="b.e21.BHIST.f09_g17.CMIP6-historical.010_v2"
./xmlchange RUN_REFDATE=2015-01-01,RUN_STARTDATE=2015-01-01
./xmlchange CLM_NAMELIST_OPTS="use_init_interp=.true.  init_interp_method='use_finidat_areas'"
./xmlchange RUN_TYPE=hybrid
```
Set the hist_nhtfrq=1 and use_c13 = T

I know this is also a problem with use_c14=T. And the c13/c14 timeseries file might effect answers as well.

### Important output or errors that show the problem

All fields are identical at step 00000, but at 01800 they start to diverge:

```
 grep RMS
 RMS ACTUAL_IMMOB                     1.9579E-11            NORMALIZED  1.1151E-04
 RMS AGNPP                            4.8354E-13            NORMALIZED  1.9943E-07
 RMS AR                               9.1612E-12            NORMALIZED  1.4726E-06
 RMS BGNPP                            3.7259E-14            NORMALIZED  1.6998E-08
 RMS CH4PROD                          6.7875E-12            NORMALIZED  1.1826E-04
 RMS CH4_SURF_AERE_SAT                4.3764E-13            NORMALIZED  1.3365E-05
 RMS CH4_SURF_AERE_UNSAT              6.5525E-14            NORMALIZED  3.6636E-05
 RMS CH4_SURF_DIFF_SAT                1.4474E-15            NORMALIZED  4.7384E-06
 RMS CH4_SURF_DIFF_UNSAT              1.5199E-16            NORMALIZED  5.5104E-07
 RMS COL_FIRE_CLOSS                   2.5753E-11            NORMALIZED  3.3462E-04
 RMS COL_FIRE_NLOSS                   2.5877E-13            NORMALIZED  1.7228E-04
 RMS CONC_O2_SAT                      5.3083E-08            NORMALIZED  8.9322E-07
 RMS CONC_O2_UNSAT                    3.1502E-08            NORMALIZED  1.8251E-08
 RMS COST_NACTIVE                     2.5476E-07            NORMALIZED  2.0829E-06
 RMS COST_NFIX                        1.0602E-17            NORMALIZED  1.7159E-15
 RMS CWDC                             2.4920E-02            NORMALIZED  4.2131E-05
 RMS CWDC_LOSS                        1.2255E-10            NORMALIZED  1.2563E-04
 RMS CWDC_vr                          3.4802E-02            NORMALIZED  4.4043E-05
 RMS CWDN                             1.1650E-11            NORMALIZED  5.0512E-12
 RMS CWDN_vr                          3.7327E-11            NORMALIZED  1.5132E-11
 RMS DEADCROOTC                       2.4056E-09            NORMALIZED  4.7772E-12
 RMS DEADCROOTN                       1.1872E-11            NORMALIZED  6.2222E-12
 RMS DEADSTEMC                        1.8613E-08            NORMALIZED  1.1603E-11
 RMS DEADSTEMN                        8.3583E-11            NORMALIZED  1.3778E-11
 RMS DENIT                            3.2733E-13            NORMALIZED  2.1733E-05
 RMS DISPVEGC                         2.4458E-08            NORMALIZED  1.1000E-11
 RMS DISPVEGN                         2.0998E-10            NORMALIZED  1.9248E-11
 RMS ER                               1.8383E-10            NORMALIZED  1.7014E-05
 RMS FAREA_BURNED                     4.2503E-14            NORMALIZED  1.4367E-04
 RMS FCH4                             7.2559E-16            NORMALIZED  3.0521E-05
 RMS FCH4TOCO2                        1.8423E-12            NORMALIZED  3.9622E-05
 RMS FCO2                             6.8548E-13            NORMALIZED  1.9570E-05
 RMS FPI                              1.2983E-06            NORMALIZED  1.3182E-06
 RMS FROOTC                           7.1096E-10            NORMALIZED  1.1243E-11
 RMS FROOTC_ALLOC                     3.7259E-14            NORMALIZED  2.0016E-08
 RMS FROOTC_LOSS                      3.9406E-13            NORMALIZED  4.3135E-09
 RMS FROOTN                           1.5078E-11            NORMALIZED  1.0160E-11
 RMS FUELC                            4.5855E-02            NORMALIZED  1.7013E-05
 RMS F_DENIT                          3.2733E-13            NORMALIZED  2.1733E-05
 RMS F_N2O_DENIT                      1.8623E-14            NORMALIZED  2.2776E-05
 RMS F_N2O_NIT                        5.1416E-19            NORMALIZED  3.3140E-08
 RMS F_NIT                            8.5693E-16            NORMALIZED  3.3140E-08
 RMS GR                               1.0077E-12            NORMALIZED  7.4924E-06
 RMS GRAINC                           7.8283E-10            NORMALIZED  1.1638E-09
 RMS GRAINN                           2.3373E-23            NORMALIZED  2.2658E-21
 RMS GROSS_NMIN                       2.1269E-14            NORMALIZED  8.3274E-08
 RMS HR                               1.8226E-10            NORMALIZED  3.9768E-05
 RMS HR_vr                            2.7296E-10            NORMALIZED  5.3141E-05
 RMS LEAFC                            2.7037E-09            NORMALIZED  6.4148E-11
 RMS LEAFC_ALLOC                      5.1420E-15            NORMALIZED  4.1487E-09
 RMS LEAFC_LOSS                       1.5023E-12            NORMALIZED  2.5368E-08
 RMS LEAFN                            9.0910E-11            NORMALIZED  8.2172E-11
 RMS LITFALL                          5.9088E-12            NORMALIZED  6.4753E-09
 RMS LITR1C                           7.5878E-04            NORMALIZED  4.0087E-05
 RMS LITR1C_vr                        1.1524E-03            NORMALIZED  4.7443E-05
 RMS LITR1N                           6.6841E-10            NORMALIZED  1.5085E-09
 RMS LITR1N_vr                        1.9552E-09            NORMALIZED  4.3164E-09
 RMS LITR2C                           1.5181E-03            NORMALIZED  1.1978E-05
 RMS LITR2C_vr                        2.3055E-03            NORMALIZED  1.4230E-05
 RMS LITR2N                           7.2931E-10            NORMALIZED  3.2359E-10
 RMS LITR2N_vr                        2.3147E-09            NORMALIZED  1.0558E-09
 RMS LITR3C                           7.5904E-04            NORMALIZED  1.3329E-05
 RMS LITR3C_vr                        1.1527E-03            NORMALIZED  1.6067E-05
 RMS LITR3N                           3.6127E-10            NORMALIZED  3.2781E-10
 RMS LITR3N_vr                        1.1417E-09            NORMALIZED  1.0694E-09
 RMS LITTERC_HR                       1.8226E-10            NORMALIZED  7.5884E-05
 RMS LITTERC_LOSS                     3.4562E-10            NORMALIZED  7.3307E-05
 RMS LIVECROOTC                       3.0458E-11            NORMALIZED  1.5922E-11
 RMS LIVECROOTN                       1.0166E-12            NORMALIZED  1.3749E-11
 RMS LIVESTEMC                        4.2963E-10            NORMALIZED  5.4985E-11
 RMS LIVESTEMN                        1.3357E-11            NORMALIZED  5.1345E-11
 RMS NACTIVE                          3.7154E-13            NORMALIZED  1.9364E-05
 RMS NACTIVE_NH4                      4.3657E-15            NORMALIZED  1.0432E-05
 RMS NACTIVE_NO3                      1.5999E-16            NORMALIZED  3.8186E-07
 RMS NAM                              3.5584E-15            NORMALIZED  4.5328E-06
 RMS NAM_NH4                          3.5944E-15            NORMALIZED  9.7271E-06
 RMS NAM_NO3                          1.5892E-16            NORMALIZED  3.8247E-07
 RMS NBP                              1.8695E-10            NORMALIZED  1.9570E-05
 RMS NDEPLOY                          3.4379E-13            NORMALIZED  1.0551E-05
 RMS NECM                             1.8348E-15            NORMALIZED  3.4987E-05
 RMS NECM_NH4                         1.8383E-15            NORMALIZED  3.7531E-05
 RMS NECM_NO3                         1.5804E-17            NORMALIZED  4.5646E-06
 RMS NEE                              1.8695E-10            NORMALIZED  2.0138E-05
 RMS NEM                              5.8137E-12            NORMALIZED  1.6923E-04
 RMS NEP                              1.8383E-10            NORMALIZED  1.9857E-05
 RMS NET_NMIN                         1.9583E-11            NORMALIZED  2.4179E-04
 RMS NFIRE                            9.4753E-15            NORMALIZED  9.9733E-05
 RMS NFIX                             3.1157E-14            NORMALIZED  1.4466E-05
 RMS NNONMYC                          3.6741E-13            NORMALIZED  2.0023E-05
 RMS NNONMYC_NH4                      3.6900E-13            NORMALIZED  2.2170E-05
 RMS NNONMYC_NO3                      3.2160E-15            NORMALIZED  1.8858E-06
 RMS NPOOL                            6.1882E-10            NORMALIZED  4.5652E-09
 RMS NPP                              9.1612E-12            NORMALIZED  1.3893E-06
 RMS NPP_GROWTH                       9.1612E-12            NORMALIZED  7.4924E-06
 RMS NPP_NACTIVE                      5.5012E-12            NORMALIZED  7.2711E-06
 RMS NPP_NACTIVE_NH4                  4.4209E-13            NORMALIZED  6.7338E-06
 RMS NPP_NACTIVE_NO3                  1.8696E-13            NORMALIZED  1.9377E-05
 RMS NPP_NFIX                         2.7712E-13            NORMALIZED  1.3403E-05
 RMS NPP_NNONMYC                      4.9367E-12            NORMALIZED  7.2460E-06
 RMS NPP_NNONMYC_NH4                  3.9491E-12            NORMALIZED  6.4514E-06
 RMS NPP_NNONMYC_NO3                  1.5371E-12            NORMALIZED  2.2225E-05
 RMS NPP_NUPTAKE                      1.0169E-11            NORMALIZED  1.1096E-05
 RMS NUPTAKE                          3.4379E-13            NORMALIZED  1.0551E-05
 RMS NUPTAKE_NPP_FRACTION             1.7209E-03            NORMALIZED  3.5615E-02
 RMS PFT_FIRE_CLOSS                   1.3028E-11            NORMALIZED  2.8993E-04
 RMS PFT_FIRE_NLOSS                   1.7545E-13            NORMALIZED  1.5848E-04
 RMS POTENTIAL_IMMOB                  1.9895E-11            NORMALIZED  1.0769E-04
 RMS POT_F_DENIT                      3.2735E-13            NORMALIZED  2.1044E-05
 RMS RETRANSN                         9.9678E-13            NORMALIZED  6.5183E-14
 RMS RR                               5.6945E-13            NORMALIZED  5.0942E-07
 RMS SLASH_HARVESTC                   8.8230E-15            NORMALIZED  1.0059E-11
 RMS SMINN                            3.5342E-08            NORMALIZED  1.5460E-09
 RMS SMINN_TO_NPOOL                   3.4379E-13            NORMALIZED  1.6110E-05
 RMS SMINN_TO_PLANT                   1.9679E-11            NORMALIZED  1.5495E-09
 RMS SMINN_TO_PLANT_FUN               3.4379E-13            NORMALIZED  1.6110E-05
 RMS SMINN_vr                         5.2653E-08            NORMALIZED  4.4009E-09
 RMS SMIN_NH4                         3.5157E-08            NORMALIZED  2.3163E-08
 RMS SMIN_NH4_vr                      5.2395E-08            NORMALIZED  2.7014E-08
 RMS SMIN_NO3                         5.9391E-10            NORMALIZED  2.7827E-11
 RMS SMIN_NO3_LEACHED                 1.3463E-17            NORMALIZED  7.0496E-09
 RMS SMIN_NO3_RUNOFF                  4.8427E-16            NORMALIZED  2.3254E-08
 RMS SMIN_NO3_vr                      1.3497E-09            NORMALIZED  1.3463E-10
 RMS SOIL1C                           2.4764E-07            NORMALIZED  2.9768E-09
 RMS SOIL1C_vr                        3.7053E-07            NORMALIZED  4.7971E-09
 RMS SOIL1N                           3.0955E-08            NORMALIZED  2.9768E-09
 RMS SOIL1N_vr                        4.1426E-08            NORMALIZED  5.3634E-09
 RMS SOIL2C                           4.5750E-08            NORMALIZED  1.8523E-11
 RMS SOIL2C_vr                        7.0063E-08            NORMALIZED  3.6995E-11
 RMS SOIL2N                           4.1591E-09            NORMALIZED  1.8523E-11
 RMS SOIL2N_vr                        5.6969E-09            NORMALIZED  4.1362E-11
 RMS SOILC_CHANGE                     1.0169E-11            NORMALIZED  1.1096E-05
 RMS SOILC_vr                         4.3996E-07            NORMALIZED  7.5212E-11
 RMS SOILN_vr                         5.2626E-08            NORMALIZED  9.8473E-11
 RMS SOM_C_LEACHED                    4.1610E-17            NORMALIZED  5.1216E-02
 RMS SR                               1.8218E-10            NORMALIZED  3.1956E-05
 RMS STORVEGC                         1.9889E-08            NORMALIZED  1.0640E-10
 RMS STORVEGN                         6.6016E-10            NORMALIZED  3.3721E-11
 RMS TOTCOLC                          2.7616E-02            NORMALIZED  1.1486E-06
 RMS TOTCOLCH4                        9.5756E-09            NORMALIZED  1.6811E-08
 RMS TOTCOLN                          7.5210E-10            NORMALIZED  3.8402E-13
 RMS TOTECOSYSC                       2.7616E-02            NORMALIZED  1.1473E-06
 RMS TOTECOSYSN                       7.5210E-10            NORMALIZED  3.8389E-13
 RMS TOTLITC                          3.0360E-03            NORMALIZED  1.4984E-05
 RMS TOTLITC_1m                       2.9757E-03            NORMALIZED  1.4804E-05
 RMS TOTLITN                          1.7478E-09            NORMALIZED  4.6007E-10
 RMS TOTLITN_1m                       1.7464E-09            NORMALIZED  4.6375E-10
 RMS TOTPFTC                          3.7336E-08            NORMALIZED  1.5666E-11
 RMS TOTPFTN                          7.4833E-10            NORMALIZED  2.4968E-11
 RMS TOTSOMC                          2.9332E-07            NORMALIZED  1.4057E-11
 RMS TOTSOMC_1m                       2.8788E-07            NORMALIZED  3.7422E-11
 RMS TOTSOMN                          3.5108E-08            NORMALIZED  1.8480E-11
 RMS TOTSOMN_1m                       3.4456E-08            NORMALIZED  4.9091E-11
 RMS TOTVEGC                          3.7336E-08            NORMALIZED  1.5490E-11
 RMS TOTVEGN                          7.4833E-10            NORMALIZED  2.4548E-11
 RMS WOODC                            2.1296E-08            NORMALIZED  1.0058E-11
 RMS WOODC_ALLOC                      6.7087E-14            NORMALIZED  4.6659E-08
 RMS WOODC_LOSS                       7.9003E-12            NORMALIZED  3.1116E-09
```

### LINKED ISSUE #686: Add a system test: turning on carbon isotopes shouldn't change answers
We should have a test that runs done with and without carbon isotopes don't change answers. This would be similar to #571 , and similar to the existing LVG test.

With the new cprnc, for this to pass, we'd need a namelist flag that turns off the carbon isotope diagnostic fields so that there are no differences in field lists.

### PR DESCRIPTION
**Pull request by ekluzek**
### Description of changes

Add a test that verifies that the code is identical when carbon isotopes are on or off. And add a fix to the bug where answers changed for land-use change cases.

### Specific notes

Contributors other than yourself, if any:@klindsey @billsacks 

CTSM Issues Fixed (include github issue #): #675 #686
 Fixes #675  
 Fixes #686

Are answers expected to change (and if so in what way)? Yes!

  Only for cases with transient land-use change AND either use_c13 and/or use_c14 turned on.

Any User Interface Changes (namelist or namelist defaults changes)? No

Testing performed, if any: Standard on hobart and nag, PASS as expected

### CODE REVIEW

File: cime_config/SystemTests/lciso.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                                       separate_builds = False,
+                                       run_two_suffix = 'cisoallon',
+                                       run_one_description = 'carbon isotopes off',
+                                       run_two_description = 'c13 and c14 isotopes on as well as C isotope time series',
+                                       ignore_fieldlist_diffs = True)
+
+    def _case_one_setup(self):
+        append_to_user_nl_files(caseroot = self._get_caseroot(),
+                                component = "clm",
+                                contents = "use_c13=F, use_c14=F")
```
Comment (billsacks) [inline cime_config/SystemTests/lciso.py]: Does this actually work? I thought you needed to set these to `.true.` or `.false.`, not `F`?

File: cime_config/testdefs/testlist_clm.xml [hunk 2]
```diff
@@ -1041,6 +1041,15 @@
       <option name="comment"  >Include one LVG debug test (exact configuration is not very important). Note that the LVG test will fail if there is any 1-d output, or output separated by glacier elevation classes (e.g., the various *_FORC fields), so this includes a testmod that turns off any 1-d output.</option>
     </options>
   </test>
+  <test name="LCISO_Lm13" grid="f10_f10_musgs" compset="IHistClm50BgcCrop" testmods="clm/ciso_monthly">
+    <machines>
+      <machine name="cheyenne" compiler="intel" category="aux_clm"/>
+    </machines>
+    <options>
+      <option name="wallclock">00:40:00</option>
+      <option name="comment"  >Make sure Carbon isotopes on and off with land-use change, does NOT change answers</option>
```
Comment (billsacks) [inline cime_config/testdefs/testlist_clm.xml]: Can you add to this comment something like, "Want test to cross the year boundary. Using _Lm13 rather than decStart due to issue #404 "

File: cime_config/SystemTests/lciso.py (hunk 1 above)
Comment (ekluzek) [reply cime_config/SystemTests/lciso.py]: Yes, I could check which version of the FORTRAN standard added it, but it is now in the standard and works on our machines and compilers and in our perl and python scripts. The one caveat there is that, in the namelist_defaults files we sometimes explicitly check for ".true." or ".false." rather than "T" or "F", but if you add the appropriate logic using the check for the IsThisTrue function it will I know I don't have the name right). The FORTRAN standard allows all of these...

.true.
.TRue.
T
F
I think it allows shortened versions as well .TRU., but not sure. But, the logical function checks for the valid forms.

File: cime_config/testdefs/testlist_clm.xml (hunk 2 above)
Comment (ekluzek) [reply cime_config/testdefs/testlist_clm.xml]: Yep, I can do that...

File: cime_config/SystemTests/lciso.py (hunk 1 above)
Comment (billsacks) [reply cime_config/SystemTests/lciso.py]: Okay, thanks, I didn't realize this. The asymmetry here (`F` and `.true.` a few lines later) still offends my sensibilities a tiny bit (I guess I'm a sensitive guy), but it's not a big deal :-)

Comment (ekluzek) [reply cime_config/SystemTests/lciso.py]: Ah, yeah I had to have that there because of the issue with build-namelist sometimes searching for the exact ".true." string. So it perhaps should be consistent. I like the being short and concise that we get with the T and F.

File: cime_config/testdefs/testlist_clm.xml (hunk 2 above)
Comment (ekluzek) [reply cime_config/testdefs/testlist_clm.xml]: Done

File: cime_config/SystemTests/lciso.py (hunk 1 above)
Comment (ekluzek) [reply cime_config/SystemTests/lciso.py]: Leaving as is. I should maybe create an issue about the build-namelist problem of doing matching on the exact ".true." string in namelist defaults.
